File: frontend/src/api/strategy.ts
import apiClient from './market'

/**
 * 策略接口
 */
export interface Strategy {
  id?: number
  name: string
  description?: string
  version?: string
  levels: string
  entryConditions: string
  exitConditions: string
  positionSizing: string
  riskControl: string
  autoTrade?: boolean
  status?: string
  createdAt?: string
  updatedAt?: string
}

/**
 * 策略 API
 */
export const strategyApi = {
  /**
   * 获取所有策略
   */
  async getStrategies(params?: {
    status?: string
    name?: string
    level?: string
  }): Promise<Strategy[]> {
    const response = await apiClient.get('/api/strategies', { params })
    return response.data
  },

  /**
   * 获取策略详情
   */
  async getStrategy(id: number): Promise<Strategy> {
    const response = await apiClient.get(`/api/strategies/${id}`)
    return response.data
  },

  /**
   * 创建策略
   */
  async createStrategy(data: Strategy): Promise<Strategy> {
    const response = await apiClient.post('/api/strategies', data)
    return response.data
  },

  /**
   * 更新策略
   */
  async updateStrategy(id: number, data: Partial<Strategy>): Promise<Strategy> {
    const response = await apiClient.put(`/api/strategies/${id}`, data)
    return response.data
  },

  /**
   * 删除策略
   */
  async deleteStrategy(id: number): Promise<void> {
    await apiClient.delete(`/api/strategies/${id}`)
  },

  /**
   * 归档策略
   */
  async archiveStrategy(id: number): Promise<Strategy> {
    const response = await apiClient.post(`/api/strategies/${id}/archive`)
    return response.data
  },

  /**
   * 获取自动交易策略
   */
  async getAutoTradeStrategies(): Promise<Strategy[]> {
    const response = await apiClient.get('/api/strategies/auto-trade')
    return response.data
  }
}

File: backend/src/main/java/com/lucance/boot/backend/entity/Strategy.java
package com.lucance.boot.backend.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.time.OffsetDateTime;

/**
 * Strategy entity for trading strategy configuration.
 * Stores entry/exit conditions, position sizing, and risk control parameters.
 */
@Data
@Entity
@Table(name = "strategies")
public class Strategy {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(length = 20)
    private String version = "1.0.0";

    /**
     * Applicable time levels for this strategy, comma-separated (e.g., "1m,5m,15m")
     */
    @Column(length = 100)
    private String levels = "1m,5m,15m";

    /**
     * Entry conditions stored as JSON string.
     */
    @Column(name = "entry_conditions", columnDefinition = "TEXT")
    private String entryConditions;

    /**
     * Exit conditions stored as JSON string.
     */
    @Column(name = "exit_conditions", columnDefinition = "TEXT")
    private String exitConditions;

    /**
     * Position sizing configuration as JSON string.
     */
    @Column(name = "position_sizing", columnDefinition = "TEXT")
    private String positionSizing;

    /**
     * Risk control parameters as JSON string.
     */
    @Column(name = "risk_control", columnDefinition = "TEXT")
    private String riskControl;

    /**
     * Whether this strategy should automatically execute trades (Phase 4 feature)
     */
    @Column(name = "auto_trade")
    private Boolean autoTrade = false;

    /**
     * Status: active, inactive, archived
     */
    @Column(length = 20)
    private String status = "active";

    @Column(name = "created_at")
    private OffsetDateTime createdAt;

    @Column(name = "updated_at")
    private OffsetDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = OffsetDateTime.now();
        updatedAt = OffsetDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = OffsetDateTime.now();
    }

    /**
     * Get levels as array.
     */
    public String[] getLevelsArray() {
        if (levels == null || levels.isBlank()) {
            return new String[0];
        }
        return levels.split(",");
    }

    /**
     * Set levels from array.
     */
    public void setLevelsArray(String[] levelsArray) {
        if (levelsArray == null || levelsArray.length == 0) {
            this.levels = "";
        } else {
            this.levels = String.join(",", levelsArray);
        }
    }
}
